pkg/deployment/deployment.go

```diff
@@ -265,24 +268,33 @@ func ConditionalDeploy(
 				readyErrorMessage,
 				err.Error()))
 			return err
-		} else if ansibleEEJob.Status.Succeeded > 0 {
+		}
+
+		if ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusSucceeded {
 			log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
 			status.Conditions.Set(condition.TrueCondition(
 				readyCondition,
 				readyMessage))
-		} else if ansibleEEJob.Status.Failed > 0 {
+			return err
+		}
+
+		if ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusRunning || ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusPending {
```

Comment:
this apparently works though I tend to prefer using else if here to make this is the continuous logic from L273.

Reply:
I did this to simplify the logic. Since we return from the previous if statement if that condition is met:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/242/files#diff-b44566f9caa976421532eb049c77b7eb247418f138516fab277a1ec33d3a7992R278

We shouldn't be able to arrive here any other way, so it didn't seem like the else was necessary in this case.

Reply:
I get your point but in that case then we probably want to return nil directly after L 287 ? My concern is that if-else logics depending on status are partially continued. If we prefer separate blocks then we probably have to make all blocks separate.

Reply:
Yeah, I guess we can. There's only one remaining unhandled condition which would be:
`	JobStatusFailed = "Failed"`

I guess my thinking there was that  the else block would also capture anything else that I wasn't already handling. But since it's just one remaining status, I think we can remove that too.

Reply:
Updated this now.